python/ray/dashboard/modules/job/job_manager.py

```diff
@@ -629,12 +629,19 @@ async def tail_job_logs(self, job_id: str) -> AsyncIterator[str]:
         if await self.get_job_status(job_id) is None:
             raise RuntimeError(f"Job '{job_id}' does not exist.")
 
+        job_finished = False
         async for lines in self._log_client.tail_logs(job_id):
             if lines is None:
-                # Return if the job has exited and there are no new log lines.
```

Comment:
We will miss logs generated between `if lines is None:` and the finish of the job.